runtime/src/main/java/org/corfudb/runtime/object/MultiVersionObject.java

```diff
@@ -327,6 +327,10 @@ private void syncStreamUnsafe(long timestamp) {
                         addressSpace.addAddress(globalAddress);
                         materializedUpTo = globalAddress;
                         resolvedUpTo = globalAddress;
+                    } catch (IllegalStateException e) {
```

Comment:
This makes the assumption that all `IllegalStateException` exceptions are retriable, but it's not true in general. Recasting the exception can be confusing. Why not just handle it as a special case? It seems like you're doing this because `Preconditions.checkState`, but do you need to use that particular method?

Reply:
I have removed the Preconditions.checkState call and replaced it with a manual check. If this regression is detected, a TrimmedException is thrown instead of the IllegalStateException. Handling proceeds as before.